php-frontend/src/main/java/org/sonar/plugins/php/api/tree/declaration/ParameterTree.java

```diff
@@ -66,4 +66,15 @@
   @Nullable
   ExpressionTree initValue();
 
+  @Nullable
+  SyntaxToken readonlyToken();
+
+  default boolean isReadonly() {
+    return readonlyToken() != null;
+  }
+
+  default boolean initProperty() {
```

Comment:
I didn't understand at first what this is. If I understood correctly, this is about the feature of adding class properties through constructor parameters. In that case, I suggest calling it `isPropertyPromotion` as it appears that this is what the construct is called (https://stitcher.io/blog/constructor-promotion-in-php-8).